File: [프로그래머스] 2022 KAKAO BLIND RECRUITMENT_주차 요금 계산.py
import math
def solution(fees, records):
    answer = []
    temp = {}
    parking = {}
    
    for record in records:
        time, car, inout = record.split()
        if car not in temp:
            temp[car] = time
        else:
            if inout == "OUT":
                parkingTime = (int(time.split(":")[0]) * 60 + int(time.split(":")[1])) - (int(temp[car].split(":")[0]) * 60 + int(temp[car].split(":")[1]))
                if car not in parking:
                    parking[car] = parkingTime
                else:
                    parking[car] += parkingTime
                del(temp[car])
                
    for remain in temp.keys():
        parkingTime = 23 * 60 + 59 - (int(temp[remain].split(":")[0]) * 60 + int(temp[remain].split(":")[1]))
        if remain not in parking:
            parking[remain] = parkingTime
        else:
            parking[remain] += parkingTime
            
    for i in parking.keys():
        totalPay = fees[1] + math.ceil(max(0, (parking[i] - fees[0])) / fees[2]) * fees[3]
        parking[i] = totalPay
        
    for i in sorted(parking.items()):
        answer.append(i[1])
        
    return answer

File: [프로그래머스] 2022 KAKAO BLIND RECRUITMENT_주차 요금 계산.js
function solution(fees, records) {
    var answer = [];
    let temp = {};
    var parking = {};
    let parkingTime = 0;
    
    for(let i of records){
        let [time, car, inout] = i.split(' ');
        if(car in temp == false){
            temp[car] = time;
        } else{
            if(inout === "OUT"){
                parkingTime = (Number(time.split(':')[0]) * 60 + Number(time.split(':')[1])) - (Number(temp[car].split(':')[0]) * 60 + Number(temp[car].split(':')[1]));
                if(car in parking === false){
                    parking[car] = parkingTime;
                } else{
                    parking[car] += parkingTime;
                }
                delete temp[car];
            }
        }
    }

    for(let i in temp){
        if(i in parking === false){
            parking[i] = 23 * 60 + 59 - (Number(temp[i].split(':')[0]) * 60 + Number(temp[i].split(':')[1]));
        } else{
            parking[i] += 23 * 60 + 59 - (Number(temp[i].split(':')[0]) * 60 + Number(temp[i].split(':')[1]));
        }
    }

    for(let i in parking){
        let parkingPay = fees[1] + Math.ceil((Math.max(0, parking[i] - fees[0])) / fees[2]) * fees[3];
        answer.push([i, parkingPay]);
    }
    return answer.sort((a, b) => a[0] - b[0]).map(v => v[1]);
}